src/openapi-generator.ts

```diff
@@ -103,7 +113,13 @@ export class OpenAPIGenerator {
     };
   }
 
-  generateComponents(): Pick<OpenAPIObject, 'components'> {
+  generateComponents(
+    config?: Pick<OpenAPIObjectConfig, 'openapi'>
+  ): Pick<OpenAPIObject, 'components'> {
+    if (config?.openapi) {
+      this.openAPIVersion = config.openapi;
+    }
+
     this.definitions.forEach(definition => this.generateSingle(definition));
```

Comment:
Actually I just thought about this. The example above is not really correct. Right now if we generate schemas once and call (for example) `generateComponents` once again we would not be "generating" them since they would be kept in memory and we would only reuse the currrent refIds. So providing a different version value won't actually work when called consecutively.